app/assets/src/components/views/SampleViewV2/ReportTable.jsx

```diff
@@ -354,10 +390,36 @@ class ReportTable extends React.Component {
   };
 
   toggleExpandGenus = ({ taxId }) => {
-    const { expandedGenusIds } = this.state;
+    const { data } = this.props;
+    const { expandedGenusIds, expandAllOpened } = this.state;
     expandedGenusIds.delete(taxId) || expandedGenusIds.add(taxId);
 
-    this.setState({ expandedGenusIds: new Set(expandedGenusIds) });
+    const newExpandedAllOpened =
+      expandedGenusIds.size === data.length ||
+      (!!expandedGenusIds.size && expandAllOpened);
```

Comment:
I don't really understand why `(!!expandedGenusIds.size && expandAllOpened)` is necessary. Can you clarify?

Reply:
It guarantees that individually closing/opening leads to the same global state.
Rewrote in a less condensed format to be more understandable.